lib/mayaHydra/hydraExtensions/mhDataProducersMayaNodeToSdfPathRegistry.cpp

```diff
@@ -15,64 +15,49 @@
 //
 
 #include "mhDataProducersMayaNodeToSdfPathRegistry.h"
+#include <pxr/base/tf/instantiateSingleton.h>
 
 PXR_NAMESPACE_USING_DIRECTIVE
+TF_INSTANTIATE_SINGLETON(MAYAHYDRA_NS_DEF::MhDataProducersMayaNodeToSdfPathRegistry);
 
 namespace MAYAHYDRA_NS_DEF {
 
-std::unique_ptr<MhDataProducersMayaNodeToSdfPathRegistry>   MhDataProducersMayaNodeToSdfPathRegistry::_instance;
-
-MhDataProducersMayaNodeToSdfPathRegistry& MhDataProducersMayaNodeToSdfPathRegistry::Get()
+/* static */
+MhDataProducersMayaNodeToSdfPathRegistry& MhDataProducersMayaNodeToSdfPathRegistry::Instance()
 {
-    if (! _instance){
-        _instance.reset(new MhDataProducersMayaNodeToSdfPathRegistry());
-    }
-    return *_instance;
+    return PXR_NS::TfSingleton<MhDataProducersMayaNodeToSdfPathRegistry>::GetInstance();
 }
 
-void MhDataProducersMayaNodeToSdfPathRegistry::Add(const MObjectHandle& objectHandle, const SdfPath& thePath)
+void MhDataProducersMayaNodeToSdfPathRegistry::Add(const unsigned long& objectHandleHashCode, const SdfPath& thePath)
 {
-    if (thePath.IsEmpty() || ! objectHandle.isValid() ) {
-        TF_CODING_WARNING("Sending an empty SdfPath or an invalid objectHandle in MhDataProducersMayaNodeToSdfPathRegistry::Add, ignoring");
+    if (thePath.IsEmpty() || 0 == objectHandleHashCode) {
+        TF_CODING_WARNING("Sending an empty SdfPath or an invalid objectHandle has code in MhDataProducersMayaNodeToSdfPathRegistry::Add, ignoring");
         return;
     }
 
-    _SdfPathByObjectHandle.insert({ objectHandle, thePath });
+    _SdfPathByHashCode.insert({ objectHandleHashCode, thePath });
 }
 
-void MhDataProducersMayaNodeToSdfPathRegistry::Remove(unsigned long& objectHandleHashCode) 
+void MhDataProducersMayaNodeToSdfPathRegistry::Remove(const unsigned long& objectHandleHashCode) 
 { 
     if (0 == objectHandleHashCode) {
         return;
     }
 
-    for (auto it = _SdfPathByObjectHandle.begin(); it != _SdfPathByObjectHandle.end(); ++it) {
-        if (it->first.hashCode() == objectHandleHashCode) {
-            _SdfPathByObjectHandle.erase(it);
-            return;
-        }
+    auto it = _SdfPathByHashCode.find(objectHandleHashCode);
+    if (it != _SdfPathByHashCode.end()) {
+        _SdfPathByHashCode.erase(it);
     }
 }
 
-void MhDataProducersMayaNodeToSdfPathRegistry::Remove(const MObjectHandle& objectHandle)
+SdfPath MhDataProducersMayaNodeToSdfPathRegistry::GetPath(const unsigned long& objectHandleHashCode) const
 {
-    if(! objectHandle.isValid()){
-        return;
-    }
-    auto it = _SdfPathByObjectHandle.find(objectHandle);
-    if (it != _SdfPathByObjectHandle.end()) { 
-        _SdfPathByObjectHandle.erase(it);
-    }
-}
```

Comment:
I think you could have kept this version and have it call `Remove(objectHandle.hashCode())` if you wanted, but I'll leave it up to you

Reply:
Thanks, sorry, I had forgotten to push the code with the typo fix. I had the ooportunity to move one class which was not dependent on maya any more since I use now the hascode and no longer the MObjectHandle so I moved it to flow viewport.